### PR TITLE
fix(compiler): forward referenced dependencies not identified as deferrable

### LINKED ISSUE #52014: Defer block doesn't chunk components inside forwardRef
### Which @angular/* package(s) are the source of the bug?

core

### Is this a regression?

No

### Description

Defer block won't chunk the components that are used with a forwardRef. 

```ts
@Component({
  selector: 'app-root',
  standalone: true,
  imports: [forwardRef(() => DeferMeCmp)],
  template: `
    <button (click)="show.set(!show())">show</button>
    <div>
      @if (show()) { 
         @defer (on immediate) {
           <defer-me />
         }
      }
    </div>
  `,
})
export class AppComponent {
  show = signal(false);
}


// defer-me.cmp.ts

@Component({
  selector: 'defer-me',
  template: ` <div>Defered!</div> `,
  standalone: true,
})
export class DeferMeCmp {}
```

### Please provide a link to a minimal reproduction of the bug

_No response_

### Please provide the exception or error you saw

_No response_

### Please provide the environment you discovered this bug in (run `ng version`)

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by crisbeto**
Fixes that we weren't accounting for dependencies using `forwardRef` when determining if they can be lazy-loaded.

Fixes #52014.

### CODE REVIEW

File: packages/compiler-cli/test/ngtsc/ngtsc_spec.ts [hunk 1]
```diff
@@ -9010,6 +9010,98 @@ function allTests(os: string) {
...
+
+          expect(jsContents).toContain('ɵɵdefer(1, 0, TestCmp_Defer_1_DepsFn)');
+          expect(jsContents).toContain('() => [import("./cmp-a").then(m => m.CmpA)]');
+
+          // The `CmpA` symbol wasn't referenced elsewhere, so it can be defer-loaded
+          // via dynamic imports and an original import can be removed.
+          expect(jsContents).not.toContain('import { CmpA }');
+        });
+
+        it('should lazy-load dependencies in an array that is not inline', () => {
```
Comment (crisbeto) [inline packages/compiler-cli/test/ngtsc/ngtsc_spec.ts:9055]: We basically get this for free by using the static evaluator.

**Review by AndrewKushnir — APPROVED** [review]
@crisbeto thanks for the fix!

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:570689999:BASE:570688959:1696430121656:a6b1d720).

**Comment by alxhub** [comment]
This PR was merged into the repository by commit 1d871c03a523e10bb838cb0f9550595cfbd9d14d.

**Pull request closed without merging**